### PR TITLE
New "<resource>_list" permission type

### PR DESCRIPTION
**Pull request by Kami**
This pull request implements new <resource>_list permission type which is used to protect "get all" API endpoints.

Keep in mind that this is stop gap for v1.0. Post v1.0, we will implement actual list filtering.

### CODE REVIEW

File: st2common/st2common/rbac/resolvers.py [hunk 1]
```diff
@@ -176,8 +217,37 @@ class SensorPermissionsResolver(PermissionsResolver):
     """
 
     def user_has_permission(self, user_db, permission_type):
-        # TODO
-        return True
+        assert permission_type in [PermissionType.SENSOR_LIST]
```
Comment (lakshmi-kannan) [inline st2common/st2common/rbac/resolvers.py]: There is a lot of code duplication here. Can you make a note in JIRA to refactor this?

Comment (Kami) [inline st2common/st2common/rbac/resolvers.py]: Sure, it was actually on my TODO, will just do it today :P

**Comment by manasdk** [comment]
Post 1.0 will we completely drop `*_list` permission? If we do keep it how will it work with `*_get`. Just want to make sure we do not create a backwards compat concern for us.

**Comment by manasdk** [comment]
I am overall +1 to the change.

**Comment by Kami** [comment]
> Post 1.0 will we completely drop *_list permission? If we do keep it how will it work with *_get. Just want to make sure we do not create a backwards compat concern for us.

We can either decide to keep it (it will simple have precedence over implicit "view" grants on a particular resources) or remove it. If we decide to remove it we can decide to provide a migration tool.

**Comment by manasdk** [comment]
Hmm, I am not convinced about keep it post 1.0 and nor I am unable to follow a reasonable migration path. But we shall see - hopefully we can entirely drop it without much of a penalty.

**Comment by Kami** [comment]
Pushed the refactor.

**Pull request merged**